### PR TITLE
[Merge Ready]Medipen middle ground, direct nerf but more versatile

### PR DESCRIPTION
**Pull request by Djiq**
## About The Pull Request 
Alternative to #51706

Solves the problem of 500 points full heal medipen with making it have c2 chems, and cutting down on what is inside of it.
Deleted Lavaland Extract
Added a luxury medipen containing c3 chems and penthrite. costs 1k points.
In pressurized enviroments it takes 10 seconds to inject and each injection only injects half the volume. Depressurized enviroments are still instant.

## Why It's Good For The Game

Middle ground is most often the better option

## Changelog
:cl:Edge and KathyRyals
tweak: Nanotrasen found out about the amount of expensive chemicals served in survival medipens and cut them down to safe costs...
add: ...And they turned the old medipen into a luxury
del: Lavaland Extract was found to break laws of physics and was removed from existance
/:cl:

cobby edit: credits/bucks > points

### CODE REVIEW

**Comment by ATH1909** [comment]
I think just the "In pressurized enviroments it takes 10 seconds to inject and each injection only injects half the volume. Depressurized enviroments are still instant." thing might be enough.

**Comment by EdgeLordExe** [comment]
> I think just the "In pressurized enviroments it takes 10 seconds to inject and each injection only injects half the volume. Depressurized enviroments are still instant." thing might be enough.

you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.

**Comment by KathyRyals** [comment]
Amazing PR, if it gets good feedback from maintainers / head coders I'll close mine.

**Comment by lovegreenstuff** [comment]
get rid of lavaland extract yes...
glory to chemistry...

nah but seriously who thought a 5 brute and 5 burn per tick chem for miners who can get the shit for it passively while they hunt for their gamer loot at the press of a button was a good idea?

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -214,13 +214,31 @@
...
 	icon_state = "stimpen"
 	inhand_icon_state = "stimpen"
+	volume = 30
+	amount_per_transfer_from_this = 30
+	list_reagents = list( /datum/reagent/medicine/epinephrine = 10, /datum/reagent/medicine/c2/aiuri = 10, /datum/reagent/medicine/c2/libital = 10)
+
+/obj/item/reagent_containers/hypospray/medipen/survival/inject(mob/living/M, mob/user)
+	if(!lavaland_equipment_pressure_check(get_turf(user)))
+		to_chat(user,"<span class='notice'>You start manually releasing the low-pressure gauge...</span>")
+		if(do_after(M,10 SECONDS))
```
Comment (Rohesie) [inline code/modules/reagents/reagent_containers/hypospray.dm:228]:
```suggestion
	if(!lavaland_equipment_pressure_check(get_turf(user)))
		if(target in user.do_afters)
			return
		to_chat(user,"<span class='notice'>You start manually releasing the low-pressure gauge...</span>")
		if(do_after(user,10 SECONDS, target = M))
```
No spamming the medipen to get both halves in one go.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 2]
```diff
@@ -214,13 +214,31 @@
...
+			amount_per_transfer_from_this = initial(amount_per_transfer_from_this)/2
```
Comment (Rohesie) [inline code/modules/reagents/reagent_containers/hypospray.dm:229]:
```suggestion
			amount_per_transfer_from_this = initial(amount_per_transfer_from_this) * 0.5
```
Prefer multiplications over divisions andusespacesplease

**Comment by Doosteroni** [comment]
>500 bucks

Let’s be clear here it’s MINING POINTS and not credits so stop wording it as so

**Comment by SkeletalElite** [comment]
> you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.

Mining using credits got reverted. You cannot get medipens with money, only mining points

**Comment by EdgeLordExe** [comment]
> > 500 bucks
> 
> Let’s be clear here it’s MINING POINTS and not credits so stop wording it as so

alrighr mining points. big fucking diffrence

**Comment by EdgeLordExe** [comment]
> > you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.
> 
> Mining using credits got reverted. You cannot get medipens with money, only mining points

i know

**Comment by ATH1909** [comment]
> > > you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.
> > 
> > 
> > Mining using credits got reverted. You cannot get medipens with money, only mining points
> 
> i know

so don't balance it around what credits can get you from a medbay vendor

**Comment by Shadowflame909** [comment]
> get rid of lavaland extract yes...
> glory to chemistry...
> 
> nah but seriously who thought a 5 brute and 5 burn per tick chem for miners who can get the shit for it passively while they hunt for their gamer loot at the press of a button was a good idea?

It's because megafauna and lava pierce through armor and do 20-50 per tick when you get hit.

Shiz balanced. This is crappy and will make people cry.

**Comment by EdgeLordExe** [comment]
> > > > you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.
> > > 
> > > 
> > > Mining using credits got reverted. You cannot get medipens with money, only mining points
> > 
> > 
> > i know
> 
> so don't balance it around what credits can get you from a medbay vendor

so you would rather i nerf orm or what? the fact that a miner can afk for 10 minutes mining stone and get 20 insta heal pens is not fucking balanced.

**Comment by Shadowflame909** [comment]
> > > > > you understand that for 500 bucks you cannot even buy an advanced medical kit and those fuckers get a 60u instant medipen with c3 chems? if that was balanced around medical those pens would be worth 3k , this is necessary.
> > > > 
> > > > 
> > > > Mining using credits got reverted. You cannot get medipens with money, only mining points
> > > 
> > > 
> > > i know
> > 
> > 
> > so don't balance it around what credits can get you from a medbay vendor
> 
> so you would rather i nerf orm or what? the fact that a miner can afk for 10 minutes mining shit and get 20 insta heal pens is not fucking balanced.

FOR THE STATION???

Why are you ignoring the lavaland circumstances that make healing of this caliber necessary.

**Comment by lovegreenstuff** [comment]
> I'm not gonna paste it all here but hi Shadowflame909 (message 1)

I'd honestly argue that it isn't. Fairly cheap, easy to abuse (amazing heal, slowdown removal, instant use, cheap price, etc.) for miners who's main job is **not technically to murder shit** but to mine. You're talking like lava and getting melee'd is inevitable, but you have a ranged weapon and lava is fucking stationary. 

> Shadowflame909 (message 2)

Healing of that caliber is not that necessary. Miners just need healing, period. It does not have to be amazing healing, just portable healing. On that note, I will say that the emergency version leans a bit too hard on the organ damage cobbychems. 10u libital will do 15 liver damage, and that's with one pen. I'd suggest a smaller amount of libital and aiuri, with minute amounts of other chems like probital. Gives the familiar feeling of an extreme burst of healing that tapers off.

Speaking from a numbers standpoint, the heal/tick of Lavaland extract is ridiculous. Guaranteed 5 brute/burn per tick heal that outclasses even Sali/Oxan combined is ridiculous. The only thing that compares is Atropine, a crit heal chem (as long as the person in question has literally every single damage type). 

This is disregarding the fact that survival medipens massively abuse chemstacking, leading to a 500 point pen that heals 9.5 brute/burn per tick as well as remove slowdowns from damage and stabilize temperature.

Finally, balancing for the station should be our priority. That's where the majority of the players will be anyways. Lavaland extract exists purely for miners and is honestly one of the reasons they're so powerful. It's insane healing on top of insane armor.

**Comment by lovegreenstuff** [comment]
Also I think it's hilarious that you can counter miner healing by stabbing them with an epipen Sseth style.

**Comment by Doosteroni** [comment]
> Healing of that caliber is not necessary

Have you ever been to lavaland? No seriously have you ever mined? Have you ever heard of these things called megafauna? I want you to try and beat megafauna with a weakened Medipen and come back.

**Comment by Shadowflame909** [comment]
Exactly, I feel like using what cobby themselves call "Medicine for minor scratches" as a replacement to counter literal instant death scenarios like Ash Storms, getting hit by a megafauna once, running into a tendril, being hit by a watcher once as a lizard.  Accidentally stepping into lava, just literally taking any damage on lavaland is bound to put you in the red. It is very ignorant of half a decade of work that went into lavaland.

It is **intentionally** balanced in such a way to make non-lavaland items null and ineffective. Otherwise, a security officer going on lavaland would be able to theoretically throttle and cheese it.

This PR spits in the face of lavaland balance by making its dangers completely unapproachable. 

You're throwing the baby out with the bath water, instead of specifically addressing why it's necessary to be able to heal so quickly, and so much. You just gut it entirely, whilst still leaving the overpowered lethality of it intact.

So, Any other PR would be a much better option than this one.

**Comment by EdgeLordExe** [comment]
LMAO yall salty af. Station is the primary place were gameplay suffers, lavaland is secondary. I dont fucking care it was balanced around lavaland. This shit breaks balance on station. Either make your own damn pr nerfing shit on lavaland or fuck off from this pr.

**Comment by Shadowflame909** [comment]
Okay...but better alternatives exist.

Why did you feel the need to make this?

**Comment by EdgeLordExe** [comment]
> Okay...but better alternatives exist.
> 
> Why did you feel the need to make this?

so you would rather medipens not work in pressurized rooms or what?

**Comment by Shadowflame909** [comment]
Unironically yeah because I feel like lavaland is too hard without them and the station doesn't need them.

**Comment by EdgeLordExe** [comment]
> Unironically yeah because I feel like lavaland is too hard without them and the station doesn't need them.


miners going murderbone do, since they can fully heal with a single fucking pen, on station , without help of medbay, and they can fill their whole backpack with them.

**Comment by Shadowflame909** [comment]
> > Unironically yeah because I feel like lavaland is too hard without them and the station doesn't need them.
> 
> miners going murderbone do, since they can fully heal with a single fucking pen, on station , without help of medbay, and they can fill their whole backpack with them.

Sounds like an issue adequately addressed by the other PRs preventing its use on station/making  it much much worse on station.

Making it much worse everywhere just hurts the mining job.

**Comment by EdgeLordExe** [comment]
> > > Unironically yeah because I feel like lavaland is too hard without them and the station doesn't need them.
> > 
> > 
> > miners going murderbone do, since they can fully heal with a single fucking pen, on station , without help of medbay, and they can fill their whole backpack with them.
> 
> Sounds like an issue adequately addressed by the other PRs preventing its use on station/making it much much worse on station.
> 
> Making it much worse everywhere just hurts the mining job.

i mean you still have luxury medipens that are arguably better than old survival medipens

**Comment by Shadowflame909** [comment]
It certainly won't save me when I'm dealing with Lavaland at roundstart with my bummy noob bait c2 chem survival pen.

And I don't think it'll be very adequate in a fight against megafauna either. Still a 🚫 from me dawg

**Comment by EdgeLordExe** [comment]
> It certainly won't save me when I'm dealing with Lavaland at roundstart with my bummy noob bait c2 chem survival pen.
> 
> And I don't think it'll be very adequate in a fight against megafauna either. Still a  from me dawg

are you telling me that this would cause miners to be more careful without thier survival insta heal and not Rush megafauna 5 minutes in? sounds good to me

**Comment by Shadowflame909** [comment]
Carefulness doesnt mean squat when bubblegum gets aggrod just from seeing you or you accidentally hit gibonite.

It's called a lavaland team wipe, and it happens a lot.

**Comment by Doosteroni** [comment]
Without effective healing meds miners will run from fauna and increase the chance of bringing megafauna to the station

**Comment by Doosteroni** [comment]
There are very few jobs that guarantee quick death if you step on 1 wrong tile or if you have slightly bad internet. We shouldn’t gimp the medipen further as it is the only thing allowing miners to get ores without dying instantly

**Comment by GoldenAlpharex** [comment]
I've said it before, and I'll say it again. If you don't want the Medipen to work on the station, then just use code to make sure that it's slower _when used on the station Z-level_. If you make it pressure-dependant, people will be penalized for going inside, where it's supposed to be safer, to heal when they're almost dead because they got hit hard by an ash storm. There's no reason to make this a pressure matter, just check if it's used on the mining z-level.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 3]
```diff
@@ -214,13 +214,33 @@
...
 	icon_state = "stimpen"
 	inhand_icon_state = "stimpen"
+	volume = 30
+	amount_per_transfer_from_this = 30
+	list_reagents = list( /datum/reagent/medicine/epinephrine = 10, /datum/reagent/medicine/c2/aiuri = 10, /datum/reagent/medicine/c2/libital = 10)
+
+/obj/item/reagent_containers/hypospray/medipen/survival/inject(mob/living/M, mob/user)
+	if(!lavaland_equipment_pressure_check(get_turf(user)))
+		if(M in user.do_afters)
+			return
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/reagent_containers/hypospray.dm:228]: feedback to `user` please

**Comment by Shadowflame909** [comment]
> 
> wyci, open your own pr if you want to change the game, i decided this is a better idea, therefore i pr'd it. tl;dr learn to code.

He says with 2 alternative PRs open...

Dare it become 3?

**Comment by Deafult-Profile** [comment]
I guess it's time to learn how to do bubblegum no hit

**Comment by Shadowflame909** [comment]
^ Cobby please no. Enough mining nerfs have been merged today.

Your making the above elitist too powerful. And this is gonna be unplayable :cry:

**Comment by Deafult-Profile** [comment]
Pr a mining buff then

**Comment by TheVekter** [comment]
> 
> 
> I've said it before, and I'll say it again. If you don't want the Medipen to work on the station, then just use code to make sure that it's slower _when used on the station Z-level_. If you make it pressure-dependant, people will be penalized for going inside, where it's supposed to be safer, to heal when they're almost dead because they got hit hard by an ash storm. There's no reason to make this a pressure matter, just check if it's used on the mining z-level.

1) Poke your head outside
2) Inject medipen
3) Go back inside

Quit griping about literally nothing.

**Comment by Shadowflame909** [comment]
Ash Storms called Vekter and they messed you up.

**Comment by ExcessiveUseOfCobblestone** [comment]
you should stop taking drugs while playing ss13 so <1 second doesn't feel like 30+ to you.

**Comment by Shadowflame909** [comment]
It's not the time. It's the damage.

Also it's noted that since hsrdsuits just lost fire immunity. These cat 2 survival pens sure won't help with temperature burn.

It's like bungee jumping with a piece of cobweb string. Its not gonna hold against the bullshit nature of lavaland.

**Comment by Coastfront** [comment]
Episode 20 of just make it only work on lavaland and do something to make it nonviable on station.

**Comment by ExcessiveUseOfCobblestone** [comment]
if you fall into orange tile you will die even if you use the emergency pen. You should probably give it a hint of leporazine (smh no c2 version of leporazine).

**Comment by EdgeLordExe** [comment]
i will just code c2 version of leporazine

**Comment by EdgeLordExe** [comment]
@ExcessiveUseOfCobblestone Added 6u of leporazine to the pen, decreased the number of other chemicals by 2u

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 4]
```diff
@@ -214,13 +214,34 @@
...
 	icon_state = "stimpen"
 	inhand_icon_state = "stimpen"
+	volume = 30
+	amount_per_transfer_from_this = 30
+	list_reagents = list( /datum/reagent/medicine/epinephrine = 8, /datum/reagent/medicine/c2/aiuri = 8, /datum/reagent/medicine/c2/libital = 8 ,/datum/reagent/medicine/leporazine = 6)
+
+/obj/item/reagent_containers/hypospray/medipen/survival/inject(mob/living/M, mob/user)
+	if(!lavaland_equipment_pressure_check(get_turf(user)))
+		if(M in user.do_afters)
+			to_chat(user,"<span class='notice'>You are already releasing the low-pressure gauge!</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/reagent_containers/hypospray.dm:228]:
```suggestion
			to_chat(user,"<span class='notice'>You are too busy to use [src]!</span>")
```

1. doafters are not specific to the pen

2. Why does the suggestion feature automatically generate a new line? who did this???

**Comment by Shadowflame909** [comment]
This was inevitable

But a better change is coming! At the steep price of 10 dollars.

**Comment by Doosteroni** [comment]
Curseth cobby for merging this fat turd.

**Comment by daplanman12** [comment]
This is a bad idea. Pens are instant. This makes it so any miner can effectively instagib any other miner with no real counterplay, thanks to the penthrite explosion reaction. 

To be quite honest, anyone in favor of this either has no idea of this very obvious interaction, or hates lavaland miners so much that they want to add a instakill for miners. If it took time to inject, or requires some degree of work to pull off, I would be in favor of this, but as is, this is just a dumb way to disincentivize shaft mining.

**Pull request merged**